### PR TITLE
#184752092: ft-reset-password

### PR DESCRIPTION
**Pull request by Tparfaite**
# What does this PR do 
implements the functionality for resetting password
# Description of the task to be completed
- Create a reset password route and controller function that generates a unique token and sends a password reset email to the user's registered email address.
- Create a route and controller function for resetting the password using the token sent to the user's email address.
- Add necessary validation checks to ensure the token is valid and not expired before allowing the password reset.
- Update the user's password in the database and log them in automatically after the password reset is successful.
# How should this be manually tested
- Clone this repository
- cd into it
- Run `npm install`
- Add .env file
- Run `npm run migrate` to add all migrations
- Run `npm run seed`
- Start the server by `npm run dev`
- In post man send post request `localhost://{PORT}/api/password/requestReset` to request reset token
- Make a post request also `localhost://{PORT}/api/password/reset-password/{token}` to reset your password
# Relevant pivotal tracker user story
#184752092

### CODE REVIEW

**Review by irakevine — APPROVED** [review]

**Review by niyontwali — CHANGES_REQUESTED** [review]
Could you please rebase your branch to develop. Your branch has a lot of conflicts with the base branch. 
Also remember to squash your commits to only one.

**Review by frankflej — APPROVED** [review]
great

**Review by jcharles921 — APPROVED** [review]
Great work Parfaite 🔥✨✨✨

**Comment by niyontwali** [comment]
Hello @Tparfaite some of the checks failed, please make sure your tests are done and complete

File: src/test/passwordReset.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const { describe, it } = require('mocha');
+
+const chai = require('chai');
+
+chai.should();
+chai.use(chaiHttp);
+// const { expect } = chai;
+
+describe('testing password reset', () => {
+  const resetToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6InBhcmZhaXRldHdhZ2lyYUBnbWFpbC5jb20iLCJpYXQiOjE2ODA0NDYxMjcsImV4cCI6MTY4MDQ0OTcyN30.T7XsdFEI0ft0xW6MFFPvWjF3mkvihgB6RToIoGD5Ryk';
```
Comment (nshutijonathan) [inline src/test/passwordReset.test.js]: load the token from `.env`

**Review by nshutijonathan — COMMENTED** [review]
Rename your commit message and always try to squash them into one or two

**Review by niyontwali — APPROVED** [review]
Well done

**Pull request merged**